lib/generators/deimos/schema_class_generator.rb

```diff
@@ -0,0 +1,231 @@
+# frozen_string_literal: true
+
+require 'rails/generators'
+require 'deimos'
+require 'deimos/utils/schema_class_mixin'
+require 'deimos/config/configuration'
+
+# Generates new schema classes.
+module Deimos
+  module Generators
+    # Generator for Schema Classes used for the IDE and consumer/producer interfaces
+    class SchemaClassGenerator < Rails::Generators::Base
+      include Deimos::Utils::SchemaClassMixin
+
+      SPECIAL_TYPES = %i(record enum).freeze
+
+      source_root File.expand_path('schema_class/templates', __dir__)
+
+      argument :full_schema,
+               desc: 'The fully qualified schema name or path.',
+               required: false,
+               type: :string
+
+      no_commands do
+        # Retrieve the fields from this Avro Schema
+        # @return [Array<SchemaField>]
+        def fields
+          @current_schema.fields.map { |field| Deimos::SchemaField.new(field.name, field.type) }
+        end
+
+        # Converts Deimos::SchemaField's to String form for generated YARD docs
+        # @param schema_field [Deimos::SchemaField]
+        # @return [String] A string representation of the Type of this SchemaField
+        def deimos_field_type(schema_field)
+          _field_type(schema_field.type)
+        end
+
+        # @param enum [Avro::Schema::EnumSchema] a field of type 'enum'.
+        # @return [Array<String>] of symbols valid for the enum.
+        def enum_symbols(enum)
+          enum.symbols
+        end
+
+        # @param enum [Avro::Schema::EnumSchema] a field of type 'enum'.
+        # @return [String] the possible return values for this Enum type
+        def enum_return_values(enum)
+          "'#{enum.symbols.join("', '")}'"
+        end
+
+        # Generate a Schema Model Class and all of its Nested Records from an Avro Schema
+        # @param schema_name [String] the name of the Avro Schema in Dot Syntax
+        def generate_classes_from_schema(schema_name)
+          _schema_base(schema_name).load_schema!
+          _schema_base.schema_store.schemas.each_value do |schema|
```

Comment:
Wondering if it would make more sense to read the Deimos config rather than the file system for this.

Reply:
I was actually thinking of that recently, i could possibly use the `Deimos.config.consumer_objects` and `Deimos.config.producer_objects` to achieve this

My quick investigating showed that this might be a bit of work to get a working solution though, but would be promising as the requirements for those objects are to set the schema and other things

Reply:
Hrmm, this may be challenging to set up with rspecs - may need to mock it out quite a bit for that.

Reply:
Gonna reserve this for last, got most of the other requested changes in!

Reply:
Looking into this for the next 15% time. Have most/all of the other comments addressed!